akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/PersistentBehaviorSpec.scala

```diff
@@ -434,6 +454,37 @@ class PersistentBehaviorSpec extends ActorTestKit with TypedAkkaSpecWithShutdown
       probe.expectMessage("msg received")
     }
 
+    "tag events" in {
+      val pid = nextPid
+      val c = spawn(counter(pid).withTagger(_ ⇒ Set("tag1", "tag2")))
+      val replyProbe = TestProbe[State]()
+
+      c ! Increment
+      c ! GetValue(replyProbe.ref)
+      replyProbe.expectMessage(State(1, Vector(0)))
+
+      val events = queries.currentEventsByTag("tag1").runWith(Sink.seq).futureValue
+      events shouldEqual List(EventEnvelope(Sequence(1), pid, 1, Incremented(1)))
+    }
+
+    "adapt events" in {
+      val pid = nextPid
+      val c = spawn(counter(pid).eventWrapper(new WrapperEventWrapper[Event]))
+      val replyProbe = TestProbe[State]()
+
+      c ! Increment
+      c ! GetValue(replyProbe.ref)
+      replyProbe.expectMessage(State(1, Vector(0)))
+
+      val events = queries.currentEventsByPersistenceId(pid).runWith(Sink.seq).futureValue
+      events shouldEqual List(EventEnvelope(Sequence(1), pid, 1, Wrapper(Incremented(1))))
```

Comment:
so if one would like the unwrapped here it could be possible to use the same adapter in a `map`?